flang/lib/Evaluate/check-expression.cpp

```diff
@@ -565,7 +566,7 @@ class CheckSpecificationExprHelper
       if (!scope_.IsModuleFile() &&
           context_.languageFeatures().ShouldWarn(
               common::LanguageFeature::SavedLocalInSpecExpr)) {
-        context_.messages().Say(
+        context_.messages().Say(common::LanguageFeature::SavedLocalInSpecExpr,
```

Comment:
Is it too soon to use Warn here and folding, or is it a matter of going through them again and updating them?

Reply:
Warn() is in Semantics and can't be used in Evaluate.